### DISCUSSION TITLE
react-hook-form How to determine in which field Enter was pressed

### DISCUSSION THREAD

**Discussion by didoshakra** [react-hook-form/react-hook-form#9813] [Q&A]
react-hook-form How to determine in which field Enter was pressed

//When entering Enter
const onKeyDown = (e) => {
  if (e.key === "Enter") {
    e.preventDefault()
    console.log("e=", e)
    if (?????.["skod"]) selectParam() // If Enter was pressed in the field
  }
}
...
<form  onSubmit={handleSubmit(onSubmit)} оnKeyDown={(e) => onKeyDown(e)}>

**Comment by leapful** [comment]
@didoshakra 

You can apply `onKeyDown` event handler on `form` element where `event.target` will be the input field name that is registered with
`useForm` hook.

```
<form onKeyDown={(event) => {
    if (event.key === 'Enter') {
        console.log(event.target.name);
    }

}} >
```

**Reply by didoshakra** [reply]
Thank you very much for your help. Your code determines in which field Enter was pressed. The code below works, although it may not work in some cases:
if (e.key === "Enter") {
        const singleValue = getValues("skod")           //The value from the required field
        //If the value of the required field is present, then we perform the necessary action
       if (singleValue != "") selParam(singleValue) 
}
